clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
@ijuma Thanks for the suggestion. Yes we can also name that new exception as `RetriableUnknownException`. And `Errors.forCode(code)` will return `RetriableUnknownException` if the code is larger than maximum code of the defined error on the client side. 

My previous idea is to name the new exception as either `UnknownException` which will be retriable. The advantage is that the error will be more explicit about that it is -- a code that is not found in the client's library. The advantage of your solution is that we can re-use the new exception for any retriable exception.

@junrao Does this sound good to you?